src/main/cliapp/src/containers/constructsPage/ConstructsTable.js

```diff
@@ -379,26 +200,40 @@ export const ConstructsTable = () => {
 			header: "Secondary IDs",
 			sortable: true,
 			filterConfig: FILTER_CONFIGS.secondaryIdsFilterConfig,
-			body: secondaryIdsBodyTemplate
+			body: (rowData) => <StringListTemplate 
+				list = {rowData.secondaryIdentifiers}
+			/>
 		},
 		{
 			field: "constructComponents.componentSymbol",
 			header: "Free Text Components",
-			body: componentsTemplate,
+			body: (rowData) => <ListDialogTemplate
+				entities={rowData.constructComponents}
+				handleOpen={handleComponentsOpen}
+				getTextField={getComponentsTextString}
+				underline={true}
+			/>,	
 			sortable: { isInEditMode },
 			filterConfig: FILTER_CONFIGS.constructComponentsFilterConfig,
 		},
 		{
 			field: "constructGenomicEntityAssociations.constructGenomicEntityAssociationObject.symbol",
 			header: "Component Associations",
-			body: genomicComponentsTemplate,
+			body: (rowData) => <ListDialogTemplate
```

Comment:
I think the `GenomicEntityListTemplate` could potentially be used for this? I think it has some of the logic you're using in the `getComponentsAssociationTextString` function.

Reply:
Yes Adam, this is similar logic and I've tried using this one, but this template cannot be used in this case.